symplyphysics/laws/thermodynamics/average_speed.py

```diff
@@ -0,0 +1,72 @@
+from sympy import Eq, sqrt, pi, integrate, S
+from symplyphysics import (
+    units,
+    Quantity,
+    Symbol,
+    print_expression,
+    validate_input,
+    validate_output,
+    symbols,
+    clone_symbol,
+)
+from symplyphysics.core.expr_comparisons import expr_equals
+from symplyphysics.laws.thermodynamics.maxwell_boltzmann_distributions import speed_distribution
+
+# Description
+## The average (mean) speed is the expected value of the speed distribution.
+
+# Law: <v> = sqrt((8 / pi) * (k * T / m))
+## <v> - average molecular speed
+## k - Boltzmann constant
+## T - equilibrium temperature
+## m - molecular mass
+
+# Conditions
+## - Assuming the particles are distributed according to Maxwell-Boltzmann statistics.
+
+average_speed = Symbol("average_speed", units.velocity, positive=True)
+equilibrium_temperature = clone_symbol(symbols.thermodynamics.temperature, "equilibrium_temperature", positive=True)
+particle_mass = clone_symbol(symbols.basic.mass, "particle_mass", positive=True)
+
+law = Eq(
+    average_speed, 
+    sqrt(8 * units.boltzmann_constant * equilibrium_temperature / (pi * particle_mass)),
+)
+
+# Derive law from Maxwell-Boltzmann distribution function using the probability formula 
+# of finding the average value of a function y(x) when the distribution function f(x) of 
+# the random variable x is known, which is calculating the integral of y(x)*f(x) with the 
+# integration limits being the lowest and highest possible values of x. In the case of 
+# this law, it is the integral of v*f(v) with v ranging from 0 to infinity.
+
+_distribution = speed_distribution.law.rhs.subs({
+    speed_distribution.equilibrium_temperature: equilibrium_temperature,
+    speed_distribution.particle_mass: particle_mass,
+})
+
+_average_speed_derived = integrate(
```

Comment:
Eventually I would like to see it as a separate law.

Reply:
You're talking about calculating the average of a function of a random variable?

Reply:
Yes.